actionpack/lib/action_controller/metal/request_forgery_protection.rb

```diff
@@ -336,5 +351,16 @@ def form_authenticity_param
       def protect_against_forgery?
         allow_forgery_protection
       end
+
+      # Checks if the request originated from the same origin by looking at the
+      # Origin header.
+      def valid_request_origin?
+        if forgery_protection_origin_check
+          # We accept blank origin headers because some user agents don't send it.
+          request.origin.blank? || request.origin == request.base_url
```

Comment:
Do they ever send a literal empty string value, or do they omit the header entirely? May be able to constrain this to a nil check.